core/trino-main/src/main/java/io/trino/split/PageSinkId.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.split;
+
+import io.trino.execution.TaskId;
+import io.trino.spi.connector.ConnectorPageSinkId;
+
+public class PageSinkId
+        implements ConnectorPageSinkId
+{
+    private final long id;
+
+    public static PageSinkId fromTaskId(TaskId taskId)
+    {
+        long stageId = taskId.getStageId().getId();
```

Comment:
can you verify that stageId, partitionId and attemptId are not outside expected ranges?

Reply:
yes. The stageId is an `int`, so we're not limiting the range there, but i'll add checks for partitionId and attemptId